File: app/src/main/java/headphone/HeadPhoneChangerReceiver.java
package headphone;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import java.util.Objects;

public class HeadPhoneChangerReceiver extends BroadcastReceiver {
    private static final String TAG = HeadPhoneChangerReceiver.class.getSimpleName();
    private final IListenesHPhoneChanger mListenes;
    public static final int PHONE_ISDICONNECTED = 0;
    public static final int PHONE_ISCONNECTED = 1;
    public static final int ERROR = -1;
    public static final String KEY_HEADSET_STATE = "state";

    public HeadPhoneChangerReceiver(IListenesHPhoneChanger mListenes) {
        this.mListenes = mListenes;
    }

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Objects.equals(intent.getAction(), Intent.ACTION_HEADSET_PLUG)) {
            Log.e(TAG, "onReceive: " + intent.getAction());
            int state = intent.getIntExtra(KEY_HEADSET_STATE, ERROR);
            if (mListenes != null) {
                mListenes.onCommand(state);//Send command some things
            }
        }
    }
}

File: app/src/main/java/android/asiantech/vn/springfinalmusic/headphone/HeadPhoneChangerReceiver.kt
package android.asiantech.vn.springfinalmusic.headphone

import android.asiantech.vn.springfinalmusic.model.Constant
import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class HeadPhoneChangerReceiver(listener: IListenerHPhoneChanger?) : BroadcastReceiver() {
    private var mListener = listener
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_HEADSET_PLUG) {
            val state = intent.getIntExtra(Constant.KEY_HEADSET_STATE, -1)
            mListener?.onCommand(state)
        }
    }

    //state=0 phone isDisconnected,state=1 phone isConnected,state =-1 ERROR
    interface IListenerHPhoneChanger {
        fun onCommand(state: Int)
    }
}
